### PR TITLE
add support for trustedProxies

### LINKED ISSUE #16897: Add trustedProxies as option for HttpsEnforcerMiddleware
### Description

HttpsEnforcerMiddleware currently has no support for setting trustedProxies.

This can be fixed by adding a `trustedProxies` array option to that will be passed to $request->setTrustedProxies([]) if it's part of the options.

### CakePHP Version

4.x

### PR DESCRIPTION
**Pull request by swiffer**
Fixes #16897

### CODE REVIEW

**Comment by othercorey** [comment]
This needs to target 4.next.

File: src/Http/Middleware/HttpsEnforcerMiddleware.php [hunk 1]
```diff
@@ -53,6 +54,7 @@ class HttpsEnforcerMiddleware implements MiddlewareInterface
         'statusCode' => 301,
         'headers' => [],
         'disableOnDebug' => true,
+        'trustedProxies' => null,
```
Comment (othercorey) [inline src/Http/Middleware/HttpsEnforcerMiddleware.php:58]: I think array should probably default to `[]`, no?

Comment (ADmad) [reply src/Http/Middleware/HttpsEnforcerMiddleware.php:58]: I would be careful with that as ServerRequest::setTrustedProxies() happily accepts empty array too.

Comment (swiffer) [reply src/Http/Middleware/HttpsEnforcerMiddleware.php:58]: to allow setting empty array (to allow every proxy) i opted for null as default to disable proxy support.

Comment (othercorey) [reply src/Http/Middleware/HttpsEnforcerMiddleware.php:58]: So empty means all? That's unfortunate.

Comment (swiffer) [reply src/Http/Middleware/HttpsEnforcerMiddleware.php:58]: yes: https://github.com/cakephp/cakephp/pull/16855/files#diff-93bf558bfa319a9d7005a51bc3fff702371b069e35abe89cb94111d906133006R80

File: src/Http/Middleware/HttpsEnforcerMiddleware.php [hunk 2]
```diff
@@ -80,6 +82,10 @@ public function __construct(array $config = [])
      */
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
+        if (is_array($this->config['trustedProxies'])) {
+            $request->setTrustedProxies($this->config['trustedProxies']);
```
Comment (ADmad) [inline src/Http/Middleware/HttpsEnforcerMiddleware.php:87]: You'll have to wrap this in a `$request instanceof ServerRequest)` check to naked the static analyzers happy.

**Review by ADmad — CHANGES_REQUESTED** [review]
Tests required for the new feature.

**Review by ADmad — APPROVED** [review]

**Pull request merged**